service/operator/service.go

```diff
@@ -387,23 +402,25 @@ func (s *Service) GetDeleteState(obj, currentState, desiredState interface{}) (i
 		{
 			newData := map[string]string{}
 			for k, v := range deleteState.ConfigMap.Data {
-				if inConfigMapData(dState.ConfigMapData, k, v) {
+				if !inConfigMapData(dState.ConfigMapData, k, v) {
```

Comment:
This changes the way the delete operation is initialized. We do not receive the zero value anymore, but the desired state, which represents the state desired to be deleted. Feels cleaner this way.